EDDI/MainWindow.xaml.cs

```diff
@@ -630,13 +652,40 @@ private void squadronRankDropDownUpdated(object sender, SelectionChangedEventArg
             }
         }
 
-        private void squadronSystemChanged(object sender, TextChangedEventArgs e)
+        // Handle changes to the editable squadron system combo box
+        private void SquadronSystemText_TextChanged(object sender, TextChangedEventArgs e)
         {
             EDDIConfiguration eddiConfiguration = EDDIConfiguration.FromFile();
-            if (eddiConfiguration.SquadronSystem != eddiSquadronSystemText.Text)
+            if (eddiConfiguration.SquadronSystem != squadronSystemDropDown.Text)
             {
-                eddiConfiguration.SquadronSystem = string.IsNullOrWhiteSpace(eddiSquadronSystemText.Text) ? null : eddiSquadronSystemText.Text.Trim();
-                eddiConfiguration = EDDI.Instance.updateSquadronSystem(eddiConfiguration);
+                string systemName = squadronSystemDropDown.Text?.ToLowerInvariant();
+                if (systemName.Length > 1)
+                {
+                    systemList = systemList.Where(s => s.StartsWith(systemName, StringComparison.InvariantCultureIgnoreCase)).ToList();
+                    if (systemList.Count < 5)
+                    {
+                        systemList = StarMapService.GetStarMapSystemsPartial(systemName)
```

Comment:
Recommended: `systemList = StarMapService.GetStarMapSystemsPartial(systemName, false, false)`
There's no need to request more information than we need, and all we really need here is names.